govuk_chat_evaluation/output_guardrails/evaluate.py

```diff
@@ -47,14 +230,41 @@ def __init__(self, evaluation_results: List[EvaluationResult]):
     def _expected_actual_lists(self) -> tuple[List[int], List[int]]:
         pairs_list = [
             (
-                int(evaluation_result.expected_outcome),
-                int(evaluation_result.actual_outcome),
+                int(evaluation_result.expected_triggered),
+                int(evaluation_result.actual_triggered),
             )
             for evaluation_result in self.evaluation_results
         ]
         expected, actual = zip(*pairs_list)
         return list(expected), list(actual)
 
+    @cached_property
+    def _expected_actual_vectors(self) -> tuple[List[List[int]], List[List[int]]]:
+        """
+        Generates lists of expected and actual binary vectors for per-guardrail evaluation.
+
+        Vectors represent triggered guardrails. If a result indicates no trigger
+        (expected_triggered or actual_triggered is False), a zero vector is used.
+        """
+        expected_vectors = []
+        actual_vectors = []
+        zero_vector = [0] * NUM_GUARDRAILS
+
+        for result in self.evaluation_results:
+            expected_vec = result.expected_exact_triggered
+            actual_vec = result.actual_exact_triggered
+
+            expected_vectors.append(
+                expected_vec if expected_vec is not None else zero_vector
+            )
+            actual_vectors.append(actual_vec if actual_vec is not None else zero_vector)
+
+        # Handle the edge case where there are no results to evaluate
+        if not expected_vectors:
+            return [], []
+
+        return expected_vectors, actual_vectors
```

Comment:
This method feels quite complex. I imagine if we always have a guardrail_parse result this could be simpler.